### ISSUE TITLE
ERRO[0003] failed to expose ports via rootlessport: "listen tcp 0.0.0.0:9080: bind: address already in use

### ISSUE THREAD

**Issue by vans163** [containers/podman#20434]
continuing.. https://github.com/containers/podman/issues/6762

```
Still a thing, I'm pretty sure that it has to do with residual binding from previous containers. This happened to me when deleting and re-creating containers multiple times. I've made sure to delete the containers but there seems to be something residual with the binding of ports left behind.

_Originally posted by @BlueOwlet in https://github.com/containers/podman/issues/6762#issuecomment-1754423518_
```

@BlueOwlet this does make sense as I was running with -rm to delete the container after exit.

**Comment by Luap99** [comment]
Please file out the bug template as requested. Without a reproducer there is nothing we can do.

**Comment by vans163** [comment]
> Please file out the bug template as requested. Without a reproducer there is nothing we can do.

A bit tied up in the coming weeks, I will try to get around to it. But its the same issue as described in the linked. Just that one was closed due to procedure.

**Comment by io7m** [comment]
I'm running into this problem repeatedly as part of the test suite for [northpike](https://github.com/io7m/northpike). The test suite starts up and shuts down various containers. It always uses `-rm`.

Something like nine runs out of ten will work perfectly, and then suddenly the tenth run hits this issue. Typically, once it has happened once, it'll happen fairly often. It seems to be next to impossible to reproduce reliably.

**Comment by Luap99** [comment]
What is your podman version?

**Comment by io7m** [comment]
```
[rm@workstation01 ~]$ podman --version
podman version 4.9.3

[rm@workstation01 ~]$ uname -a
Linux workstation01 6.7.8-arch1-1 #1 SMP PREEMPT_DYNAMIC Sun, 03 Mar 2024 00:30:36 +0000 x86_64 GNU/Linux
```

**Comment by eplightning** [comment]
Still happens with Podman 5.1.1 . Got this error randomly in CI environment where containers are being created and removed very often.
```
rootlessport listen tcp 0.0.0.0:40161: bind: address already in use
```

**Comment by eplightning** [comment]
If I'm not wrong this is used to select random free port: https://github.com/containers/podman/blob/main/utils/ports.go#L10
This is probably very prone to race conditions when starting containers at the same time.

Some kind of locking, maybe similar to what is being done to machine ports would prevent this: https://github.com/containers/podman/blob/main/pkg/machine/ports/ports.go

**Comment by Luap99** [comment]
> If I'm not wrong this is used to select random free port: https://github.com/containers/podman/blob/main/utils/ports.go#L10 This is probably very prone to race conditions when starting containers at the same time.
> 
> Some kind of locking, maybe similar to what is being done to machine ports would prevent this: https://github.com/containers/podman/blob/main/pkg/machine/ports/ports.go

This is super complicated and not possible with the current design really. ref https://github.com/containers/podman/issues/10205#issuecomment-1010055023

If your issue that you let port be auto assigned it is a dup of https://github.com/containers/podman/issues/10205

But this issue here has no description other than a error nor any meaningful reproducer for me so there is really nothing I can do about it.

**Comment by eplightning** [comment]
Yes I'm using random ports so I guess its already tracked in #10205
